apps/labs/posts/conda-on-colaboratory.md

```diff
@@ -0,0 +1,109 @@
+## What is condacolab, and how does it work?
+
+[Condacolab]((https://github.com/conda-incubator/condacolab)) is a tool that lets you deploy a Miniconda installation easily on Google Colab notebooks and use conda or mamba in colab.
+
+---
+
+![Condacolab's previous installation](/posts/conda-on-colaboratory/previous_condacolab_installation.png)
+
+---
+
+The way it worked before was by installing the Miniconda distribution on top of the system's Python at `/usr/local`, and then adding a few configuration files to ensure we stay with Python 3.7. Finally, we wrapped the Python executable to redirect and inject some environment variables needed to load the new libraries installed by `conda`. Since we need to re-read `LD_LIBRARY_PATH`, we issued a Jupyter kernel restart.
+
+The problem with this approach is that we are still stuck with Python 3.7 and overwriting the system's Python executable. This is not the best way because we are leaving some original Colab libraries that depend on other packages, resulting in a chaotic mixture of conda-provided files with colab-preinstalled files. In most simple cases, this doesn't present much of a problem. However, if users rely on more complex packages with compiled dependencies, chances are you would often run into difficult to debug errors due to ABI incompatibilities. 
+
+## Adopting a better solution 🥳
+
+The first goal we worked on during the internship was to design and implement a more robust solution to use conda and/or mamba on colab. 
+
+After some iterations, we settled for the following approach:
+
+1. We install the Miniconda distribution (or any other distribution) at `/opt/conda` instead of `/usr/local`.
+2. We supplement the `base` environment with colab-required packages, like `google-colab`, `condatools`, `psutil`, and `matplotlib`. 
+3. We overwrite `/usr/local/python` (the executable run by the default ipykernel) with a shell script that activates conda and starts our custom ipykernel, forwarding the calls there. Thanks to this step, the Jupyter server didn't even notice, but we are now running the conda Python without touching the system one at all!
+
+--- 
+
+![bash script which activates the conda base environment](/posts/conda-on-colaboratory/bash_script_activating_env.png)
+
+---
+
+---
+
+![colab's screenshot when using the new approach where we are activating the conda base environment using the python executable.](/posts/conda-on-colaboratory/colab-when-activating-base-env.png)
+
+---
+
+## Some other work we  for condacolab
+
+### 1. Adding a `Restart kernel` button.
+
+During the installation of condacolab the kernel is restarted automatically. This could make users feel like something is wrong with the installation or Colab. We now added a button to restart the kernel to resolve this issue. If you would set the `restart_kernel` to `False` during the installation, then the kernel will not restart automatically, and a button will appear, which you can click to restart the kernel.
```

Comment:
```suggestion
During the installation of condacolab the kernel is restarted automatically. This could make users feel like something is wrong with the installation or Colab. We now added a button to restart the kernel to resolve this issue. If you set `restart_kernel` to `False` during the installation, then the kernel will not restart automatically, and a button will appear, which you can click to restart the kernel.
```